test/system/departments_test.rb

```diff
@@ -0,0 +1,158 @@
+require 'application_system_test_case'
+
+class DepartmentsTest < ApplicationSystemTestCase
+  def setup
+    @user = create :user, :with_profile, :with_clients,
+      :with_departments, role: 'superadmin'
+    @user_as_social_worker = create :user, :with_profile, :with_clients,
+      :with_departments, role: 'social_worker'
+  end
+
+  test 'superadmin should see departmens link in navigation' do
+    login_as @user
+    visit root_path
+    assert page.has_link? 'Departments'
+  end
+
+  test 'other users should not see departmens link in navigation' do
+    login_as @user_as_social_worker
+    visit root_path
+    refute page.has_link? 'Departments'
+  end
+
+  test 'superadmin can see all departments in departmens_path' do
+    login_as @user
+    visit departments_path
+    Department.all.each do |d|
+      assert page.has_text? d.contact.org_role
+      assert page.has_link? 'Show', href: department_path(d.id)
+      assert page.has_link? 'Edit', href: edit_department_path(d.id)
+      assert page.has_link? 'Delete', href: department_path(d.id)
+    end
+  end
+
+  test 'superadmin can create department' do
+    login_as @user
+    visit departments_path
+    click_link 'New department'
+
+    assert page.has_field? 'Department name'
+    fill_in 'Department name', with: 'Bogus Hog Department'
+
+    assert page.has_select? 'User'
+    within '.department_user' do
+      User.department_assocable.each do |u|
+        select u.email
+      end
+    end
+
+    assert page.has_field? 'Department name'
+    fill_in 'Department', with: 'bogus Department'
```

Comment:
The `assert page.has_field? 'Department name'` is not needed, as this line should fail otherwise